### PR TITLE
Add steps for customizing spark opearator

### PR DESCRIPTION
**Pull request by dharmeshkakadia**

### CODE REVIEW

**Comment by dharmeshkakadia** [comment]
signed CLA.

**Comment by dharmeshkakadia** [comment]
ping !

File: docs/user-guide.md [hunk 1]
```diff
@@ -451,3 +452,16 @@ be used with a `ScheduledSparkApplication`. In most cases, a restart policy of `
 the next run usually picks up where the previous run left anyway. For these reasons, it's often the right choice to use
 a restart policy of `Never` as the example above shows. 
  
+## Customizing Spark Operator
+
+To customize spark operator, you need to go through following steps:
```
Comment (liyinan926) [inline docs/user-guide.md]: nit: We use `the Spark Operator` consistently in the doc. Also please add `the` before `following` and some places below, e.g., `the FROM tag`.

**Comment by dharmeshkakadia** [comment]
How does it look now @liyinan926 ?

**Comment by liyinan926** [comment]
LGTM. Thanks!

File: docs/user-guide.md [hunk 2]
```diff
@@ -451,3 +452,16 @@ be used with a `ScheduledSparkApplication`. In most cases, a restart policy of `
...
+
+1. Compile Spark distribution with Kubernetes support as per [Spark documentation](https://spark.apache.org/docs/latest/building-spark.html#building-with-kubernetes-support).
+
+2. Create docker images to be used for Spark with [docker-image tool](https://spark.apache.org/docs/latest/running-on-kubernetes.html#docker-images).
+
+3. Create a new Spark Operator image based on the above image. You need to modify the `FROM` tag in the [Dockerfile](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/Dockerfile) with your Spark image.
+
+4. Build and push your Spark Operator image built above. 
+
+5. Deploy the new image by modifying the [/manifest/spark-operator.yaml](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/manifest/spark-operator.yaml) file and specfiying your Spark Operator image.
```
Comment (indianmifan) [inline docs/user-guide.md:467]: can you add more detail about how to deploy the new image to k8s?

**Pull request merged**